src/java/io/bazel/rulesscala/specs2/Specs2RunnerBuilder.scala

```diff
@@ -0,0 +1,52 @@
+package io.bazel.rulesscala.specs2
+
+import java.util.regex.Pattern
+
+import io.bazel.rulesscala.test_discovery.{FilteredRunnerBuilder, PrefixSuffixTestDiscoveringSuite}
+import org.junit.runner.notification.RunNotifier
+import org.junit.runner.{Description, RunWith, Runner}
+import org.junit.runners.Suite
+import org.junit.runners.model.RunnerBuilder
+import org.specs2.control.Action
+import org.specs2.main.{Arguments, CommandLine, Select}
+import org.specs2.specification.core.Env
+import org.specs2.specification.process.Stats
+
+import scala.collection.JavaConverters._
+
+@RunWith(classOf[Specs2PrefixSuffixTestDiscoveringSuite])
+class Specs2DiscoveredTestSuite
+
+class Specs2PrefixSuffixTestDiscoveringSuite(testClass: Class[Any], builder: RunnerBuilder)
+  extends Suite(new Specs2RunnerBuilder(builder), PrefixSuffixTestDiscoveringSuite.discoverClasses())
+
+class Specs2RunnerBuilder(builder: RunnerBuilder) extends FilteredRunnerBuilder(builder) {
+  override def runnerFilter(runner: Runner, testClass: Class[_], pattern: Pattern): Runner = runner match {
+    case _: org.specs2.runner.JUnitRunner => new Specs2ClassRunner(testClass, pattern)
+    case _ => super.runnerFilter(runner, testClass, pattern)
+  }
+}
+
+class Specs2ClassRunner(testClass: Class[_], pattern: Pattern)
+  extends org.specs2.runner.JUnitRunner(testClass) {
+
+  private def filtered(pattern: Pattern, children: List[Description]): List[Description] =
+    children.flatMap { c => c :: filtered(pattern, c.getChildren.asScala.toList) }
+      .filter(d => {
+        val testCase = d.getClassName + "#" + d.getMethodName
+        pattern.matcher(testCase).matches
+      })
+
+  override def runWithEnv(n: RunNotifier, env: Env): Action[Stats] = {
+    val fragments = filtered(pattern, this.getDescription.getChildren.asScala.toList)
+      .map(d => d.getMethodName.split("::").reverse.headOption.getOrElse(d.getMethodName))
```

Comment:
Again, recommend adding a declarative method for clarity.
Something like 
```
val testChildren = this.getDescription.getChildren.asScala.toList
val testCases = filtered(requestedTestFilter, testChildren)
val fragments = methodNamesOf(testCases)
```

WDYT?